projects/firmware_homework/src/main.c

```diff
@@ -0,0 +1,79 @@
+// MidSun libraries
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// Interrupt callback function
+static void prv_gpio_callback(const GpioAddress *address, void *context) {
+  // Context should be the address of A6 as we want to read from this pin
+  GpioAddress *adc_reading = context;
+  uint16_t adc_data = 0;
+  // Read the values from the ADC channel and output the reading
+  // Check the data to see if it is valid first
+  if (adc_read_converted_pin(*adc_reading, &adc_data) == STATUS_CODE_OK) {
+    LOG_DEBUG("ADC Reading: %i\n", adc_data);
+  } else {
+    LOG_DEBUG("Invalid Reading: %i\n");
```

Comment:
This is failing to compile in CI because you don't pass an argument to fill the `%i` in the format string. Pass an argument or remove the `%i` to fix that and get CI to pass.